node/tests/checkers.go

```diff
@@ -0,0 +1,50 @@
+package tests
+
+import (
+	"bytes"
+	"errors"
+	"github.com/ipld/go-ipld-prime/codec/dagjson"
+
+	qt "github.com/frankban/quicktest"
+
+	"github.com/ipld/go-ipld-prime/datamodel"
+)
+
+// NodeContentEquals checks whether two nodes have equal content by first encoding them via
+// dagjson.Encode, then checking that the generated encodings are identical.
+//
+// See: dagjson.Encode.
+var NodeContentEquals = &nodeContentEqualsChecker{}
+
+type nodeContentEqualsChecker struct{}
+
+func (n *nodeContentEqualsChecker) Check(got interface{}, args []interface{}, note func(key string, value interface{})) error {
+	want := args[0]
+	if want == nil {
+		return qt.IsNil.Check(got, args, note)
+	}
+	wantNode, ok := want.(datamodel.Node)
+	if !ok {
+		return errors.New("this checker only supports checking datamodel.Node values")
+	}
+	var buf bytes.Buffer
+	if err := dagjson.Encode(wantNode, &buf); err != nil {
```

Comment:
I think I'd suggest replacing this (and the other one below) with [`printer.Sprint(n)`](https://github.com/ipld/go-ipld-prime/blob/b9a89e847312334af141788b303fd78b4f076fe3/printer/printer.go#L23).

It's going to be behaviourally awfully similar in all the key goals, but I think often better:
- better on readability (the printer produces multiline output),
- will sometimes give slightly more coverage (it's going to tell you if there's schema types involved, and what they are, if applicable),
- and it just plain gives me a bit less fear about the dependency tree getting kinky than depending on a codec package here.

Reply:
I had that implemented at first but unfortunately tests hit the unimplemented `TODO`s in printer which is a blocker.

Reply:
Another alternative is to use `datamodel.DeepEqual` in the checker and call the checker `NodeDeepEquals`

Reply:
:facepalm: :facepalm: :facepalm: 

Those might be fast enough to fill in that it's worth it.  I can try to take a look later, or maybe talk you into it?

Reply:
IMO this is precisely the kind of thing that could be a small follow-up PR :)

Reply:
Happy to be talked into stuff 🚀

Reply:
For reference, `printer` TODOs that tests hit are now implemented [here](https://github.com/ipld/go-ipld-prime/pull/294). Once that's merged I'll revert back to using `printer` to implement the checker here. 🍻

Reply:
Reverting back to using dagjson, since printer panics, caused by a TODO in test generator; see #295. Two things need to happen before we can use `printer` for the checker:

1. Implement TODO in test generator, which #295 should address
1. Be more defensive in `printer` implementation and handle `nil`s gracefully, captured in #296

The key thing in all of this is to make sure checker used in tests is correct since tests rely on it to assert content equality.